### PR TITLE
Remove/fix all references to undefined Kconfig symbols in Kconfig* files

### PR DESCRIPTION
**Pull request by ulfalizer**
This series removes/fixes all references to undefined symbols within `Kconfig*` files in Zephyr. The plan is to turn undefined references into an error later.

Additional cleanup might be possible here.

When fixing the two malformed `default n` properties, I removed them instead. Kconfig `bool` symbols implicitly default to `n`, which might be poorly documented. Please complain if you prefer to keep the `default`s, e.g. for consistency with surrounding code.

@MaureenHelm: could you check commit 88ad0dc57028 ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") and see if the symbols that those `default`s were on can be removed?

### CODE REVIEW

**Comment by ulfalizer** [comment]
CC @nashif @carlescufi @MaureenHelm @jhedberg 

No hurry if you're busy with the release.

File: subsys/net/ip/Kconfig.rpl [hunk 1]
```diff
@@ -232,7 +232,6 @@ config	NET_RPL_PREFERENCE
 config	NET_RPL_GROUNDED
 	bool "DAG grounded default value"
 	depends on NET_RPL
-	default false
```
Comment (MaureenHelm) [inline subsys/net/ip/Kconfig.rpl]: It looks this change goes with the previous commit's message, and vice versa?

**Review by MaureenHelm — COMMENTED** [review]
> @MaureenHelm: could you check commit 88ad0dc ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") and see if the symbols that those defaults were on can be removed?

LGTM

File: subsys/net/ip/Kconfig.rpl (hunk 1 above)
Comment (ulfalizer) [reply subsys/net/ip/Kconfig.rpl]: Oops... git slip-up. Fixed.

**Comment by ulfalizer** [comment]
> @MaureenHelm: could you check commit 88ad0dc ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") and see if the symbols that those defaults were on can be removed?
> 
> LGTM

Removed now

File: ext/lib/mgmt/mcumgr/cmd/log_mgmt/Kconfig [hunk 2]
```diff
@@ -18,7 +18,6 @@
 menuconfig MCUMGR_CMD_LOG_MGMT
     bool
     prompt "Enable mcumgr handlers for log management"
-    depends on MDLOG
```
Comment (ulfalizer) [inline ext/lib/mgmt/mcumgr/cmd/log_mgmt/Kconfig]: As a note here, [MCUMGR_CMD_LOG_MGMT](http://docs.zephyrproject.org/reference/kconfig/CONFIG_MCUMGR_CMD_LOG_MGMT.html) (*Enables mcumgr handlers for log management*) will have been impossible to enable until this gets in, from the moment it got merged.

**Comment by ulfalizer** [comment]
CC @galak

**Comment by MaureenHelm** [comment]
> @MaureenHelm: could you check commit 88ad0dc ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") and see if the symbols that those defaults were on can be removed?
> LGTM
> 
> Removed now

@ulfalizer I misunderstood your question before. The defaults can be removed, but I think @tbursztyka or @jfischer-phytec-iot may still need the symbols they were on. In other words, your original commit 88ad0dc ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") looked good, but I'm not sure about commit 7112c670df1727acc69bb5ed01700c0e8e6603b7

**Comment by ulfalizer** [comment]
@MaureenHelm 
Changed it to just remove the defaults. See if it looks alright now.

**Comment by ulfalizer** [comment]
@tbursztyka 
Could you check if https://github.com/zephyrproject-rtos/zephyr/pull/8106/commits/516e602bcbbc8fd5445ac80ce65ca8e271fdb3df looks okay?

**Comment by ulfalizer** [comment]
@tbursztyka
Could you check if https://github.com/zephyrproject-rtos/zephyr/pull/8106/commits/5fff025b7b7010124a7ca62e43cf08c135b41dea looks okay?

**Comment by ulfalizer** [comment]
*Ping*

**Review by nashif — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
looks good, thanks for the cleanup!

**Pull request merged**